docfiles/pxtweb/cookieCompliance.ts

```diff
@@ -108,34 +141,44 @@ namespace pxt {
         }
         export function report(filter: string = null) {
             perfReportLogged = true;
-
             if (enabled) {
-                const milestones: {[index: string]: number} = {};
-                const durations: {[index: string]: number} = {};
+                const milestones: { [index: string]: number } = {};
+                const durations: { [index: string]: number } = {};
 
-                let report = `performance report:\n`
-                for (let [msg, time] of stats.milestones) {
+                let report = `Performance Report:\n`
+                report += `\n`
+                report += `\tMilestones:\n`
+                for (let [msg, time, params] of stats.milestones) {
                     if (!filter || msg.indexOf(filter) >= 0) {
                         let pretty = prettyStr(time)
-                        report += `\t\t${msg} @ ${pretty}\n`
+                        report += `\t\t${msg} @ ${pretty}`
+                        for (let k of Object.keys(params || {})) {
+                            report += `\n\t\t\t${k}: ${params[k]}`
+                        }
+                        report += `\n`
                         milestones[msg] = time;
                     }
                 }
 
                 report += `\n`
-                for (let [msg, start, duration] of stats.durations) {
+                report += `\tMeasurements:\n`
+                for (let [msg, start, duration, params] of stats.durations) {
                     let filterIncl = filter && msg.indexOf(filter) >= 0
                     if ((duration > 50 && !filter) || filterIncl) {
                         let pretty = prettyStr(duration)
                         report += `\t\t${msg} took ~ ${pretty}`
                         if (duration > 1000) {
                             report += ` (${prettyStr(start)} - ${prettyStr(start + duration)})`
+                            for (let k of Object.keys(params || {})) {
+                                report += `\n\t\t\t${k}: ${params[k]}`
+                            }
                         }
                         report += `\n`
                     }
                     durations[msg] = duration;
                 }
                 console.log(report)
+                enabled = false; // stop collecting milestones and measurements after report
```

Comment:
I made this change because I believe it is undesirable to continue collecting perf measurements after the perf report is generated. These measurements accumulate in memory over time and never flush, which is bad if left unbounded (which it was). These metrics are most useful for gauging startup performance, so I think we don't want to continue collecting them once the editor is fully loaded. Let me know if I'm mistaken!

Reply:
If I understand correctly, these are pretty generic methods. I worry someone might want to use them down the line (to measure issues with perf when hitting the play button, for example) and get confused as to why they're not working.

Maybe we could have a flag to control this behavior?

Reply:
@thsparks I agree it could be useful beyond startup. Supporting this would require coming up with a flushing strategy beyond what exists today. That's out of scope of this change.